code/modules/surgery/organs/organ_internal.dm

```diff
@@ -362,6 +363,38 @@
 	owner.SetEyeBlurry(0)
 	owner.SetEyeBlind(0)
 
+/obj/item/organ/internal/robotize()
+	if(!(status & ORGAN_ROBOT)) //Don't override the icons for the already-mechanical IPC organs.
+		var/list/states = icon_states('icons/obj/surgery.dmi') //Insensitive to specially-defined icon files for species like the Drask or whomever else. Everyone gets the same robotic heart.
+		if(slot == "heart" && ("[slot]-prosthetic-on" in states) && ("[slot]-prosthetic-off" in states)) //Give the robotic heart its robotic heart icons if they exist.
+			var/obj/item/organ/internal/heart/H = src
+			H.icon = icon('icons/obj/surgery.dmi')
+			H.icon_base = "[slot]-prosthetic"
+			H.dead_icon = "[slot]-prosthetic-off"
+			H.update_icon()
+		else if("[slot]-prosthetic" in states) //Give the robotic organ its robotic organ icons if they exist.
+			icon = icon('icons/obj/surgery.dmi')
+			icon_state = "[slot]-prosthetic"
+		name = "mechanical [slot]"
+	..() //Go apply all the organ flags/robotic statuses.
+
+/obj/item/organ/internal/eyes/robotize()
+	colourmatrix = null
+	dark_view = 2
+	..() //Make sure the organ's got the robotic status indicators before updating the client colour.
+	owner.update_client_colour(0) //Since both mechassisted and mechanical eyes give dark_view of 2 and full colour vision atm, just having this here is fine as mechassist() will call it anyway.
+
+/obj/item/organ/internal/mechassist()
+	..() //Go back, call robotize(), adjust the robotic status indicators and the organ damage parameters.
```

Comment:
Now that `robotize` overrides the icon, an organ with a custom icon could lose it before this proc gets a chance to check for assisted organ states. When `organ/mechassist` calls `robotize`, something should prevent it from touching the icon. You could probably add a new arg to `internal/robotize`, and it shouldn't interfere with the one on `external/robotize`, since only internal organs can be mechanically assisted anyway.